app/src/modals/SwitchNetworkModal.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+import styled from 'styled-components'
+import Modal, { Footer, Props as ModalProps, RejectReason } from '../shared/components/Modal'
+import PngIcon from '../shared/components/PngIcon'
+import Buttons from '$shared/components/Buttons'
+import { ethereumNetworks } from '../shared/utils/constants'
+
+interface Props extends Pick<ModalProps, 'onReject'> {
+    expectedNetwork: number | string
+    actualNetwork: number | string
+    onResolve?: () => void
+}
+
+function getChainName(chainId: number | string) {
+    return ethereumNetworks[chainId] || `#${chainId}`
```

Comment:
There's `getConfigForChain ($shared/web3/config.ts)` which will read chain config from `@streamr/config`

Reply:
Oh, but chain name is not part of the config. We should really add that :D

Reply:
Yep, right. There's a ticket about cleaning up them chain configs. Let's address separately.